jobs/databaserunner.go

```diff
@@ -0,0 +1,147 @@
+// the endpoint and endpoint template are not currently validated but this can/should be done in the future.  If a
+// a token is passed, it will be configured but is not required.
+func NewDatabaseRunner(config map[string]interface{}) (*DatabaseRunner, error) {
+	log.Error("creating new database job runner")
+
+	var endpoint string
+	if v, ok := config["endpoint"].(string); ok {
+		endpoint = v
+	}
+
+	var endpointTemplate string
+	if v, ok := config["endpointTemplate"].(string); ok {
+		endpointTemplate = v
+	}
+
+	if endpoint == "" && endpointTemplate == "" {
+		return nil, errors.New("endpoint or endpoint_template is required")
+	} else if endpoint != "" && endpointTemplate != "" {
+		log.Warn("both endpoint and endpoint_template are set, only endpoint will be used")
+	}
+
+	var token string
+	if v, ok := config["token"].(string); ok {
+		token = v
+	}
+
+	return &DatabaseRunner{
+		Endpoint:         endpoint,
+		EndpointTemplate: endpointTemplate,
+		Token:            token,
+	}, nil
+}
+
+// Run executes the DatabaseRunner.  The instance_id and database_action are required.  Allowable actions
+// are 'start' and 'stop'.  If an endpoint is configured on the runner, it will be used, otherwise
+// we assume there is an endpointTemplate and try to execute it.  Database actions are currently only
+// executed with the PUT method and a body containing {"state": "action"}.
+func (r *DatabaseRunner) Run(ctx context.Context, account string, parameters interface{}) (string, error) {
+	if account == "" {
+		return "", errors.New("account is required")
+	}
+
+	log.Infof("running database runner %+v in account %s,  with parameters %+v", r, account, parameters)
+
+	params, ok := parameters.(map[string]string)
+	if !ok {
+		return "", NewRunnerError(ErrMissingDetails, "wrong type", errors.New("parameters list is not a map[string]string"))
+	}
+
+	instanceID, ok := params["instance_id"]
+	if !ok {
+		return "", NewRunnerError(ErrMissingDetails, "missing instance_id", nil)
+	}
+
+	action, ok := params["database_action"]
+	if !ok {
+		return "", NewRunnerError(ErrMissingDetails, "missing database_action", nil)
+	}
+
+	endpoint := r.Endpoint
+	if endpoint == "" {
+		log.Debugf("endpoint is empty, attempting to use endpoint template '%s'", r.EndpointTemplate)
+
+		input := struct {
+			Account    string
+			InstanceID string
+		}{account, instanceID}
+
+		tmpl, err := template.New("endpoint").Parse(r.EndpointTemplate)
+		if err != nil {
+			return "", NewRunnerError(ErrPreExecFailure, "template parsing failed", err)
+		}
+
+		var out bytes.Buffer
+		if err := tmpl.Execute(&out, &input); err != nil {
+			return "", NewRunnerError(ErrPreExecFailure, "template execution failed", err)
+		}
+
+		endpoint = out.String()
+		log.Debugf("parsed endpoint template: %s", endpoint)
+	}
+
+	switch action {
+	case "stop", "start":
+		j, err := json.Marshal(map[string]string{"State": action})
```

Comment:
this is really minor, but does the key need to be capitalized?  I'd rather see all of those types of things lowercased.

Reply:
I was going off this: https://github.com/YaleSpinup/rds-api/blob/e6ca0b89aa5abd2d6f4f8a5eeb3a43e218d6e38b/README.md#stopping-and-starting-a-databasecluster

Reply:
Just try hitting it with postman?  If it supports lowercase, let's update the readme in the rds-api too :)

Reply:
The buffalo doc says it uses the standard library json encoding/decoding, so I think it should work:

https://gobuffalo.io/en/docs/bind#json-and-xml-binding